### ISSUE TITLE
Collator's `CaseLevel` option not working

### ISSUE THREAD

**Issue by jedel1043** [unicode-org/icu4x#2884]
The following snippet panics:

```Rust
use icu_collator::{CaseLevel, Collator, CollatorOptions, Strength};
use icu_locid::locale;

fn main() {
    let data = icu_testdata::unstable();
    let collator = Collator::try_new_unstable(&data, &locale!("es").into(), {
        let mut opts = CollatorOptions::new();
        opts.strength = Some(Strength::Primary);
        opts.case_level = Some(CaseLevel::On);
        opts
    }).unwrap();
    let target = "Aa";
    let input = ["Aa", "aA", "áÁ", "Aã"];

    let matches = input.into_iter().filter(|s| collator.compare(s, target).is_eq()).collect::<Vec<_>>();

    assert_eq!(&matches, &["Aa, Aã"])
}
```

Variable `matches` contains `["Aa", "aA", "áÁ", "Aã"]`, which is incorrect since the `CaseLevel::On` option should make "Aa" not equal to "aA/áÁ".

**Comment by sffc** [comment]
CC @hsivonen @echeran Can you take a look?

**Comment by hsivonen** [comment]
We should probably do a 1.0.1 release of the collator with this fix.

**Comment by hsivonen** [comment]
Thanks for catching this. As you probably already noticed, the fix is now released in `icu_collator` 1.0.1.

**Comment by jedel1043** [comment]
Thank you for the quick fix! The remaining failing tests from `Intl.Collator` are more related to locale resolution than the collator itself, but I'll track the updates on the test262 suite in case they add a new test that finds a new bug on the collator.
